addons/uh60_weapons/functions/fnc_fired.sqf

```diff
@@ -16,4 +16,84 @@ if (_ammo == "VTX_Hellfire_AGM114K" || _ammo == "VTX_Hellfire_AGM114N") then {
 
 if (player == _gunner && local _gunner) then {
 	[_vehicle] call vtx_uh60_weapons_fnc_updateMFDValues;
-};
\ No newline at end of file
+};
+
+//spaggetti code:  Mix of Yax's ITC Air APKs and HARM Guidence by Donov
+if (_ammo == "ammo_Missile_HARM_HL" && vehicle _gunner == _vehicle) then {
+	_index = 0;	//reset index to enter while loop
+	while {_index < 4} do {
+		if ((getPylonMagazines _vehicle # _index) == "PylonMissile_Missile_HARM_x1_HL") then {
+			if (_vehicle ammoOnPylon (_index + 1) == 0) then { _vehicle setPylonLoadout [(_index + 1), "", true]; /*systemChat "Reset Pylon"*/};
+			//This loop checks for which HARM was fired, and replaces the pylon with nothing.  It removes the missile
+		};
+		_index = _index + 1;  //python be like
```

Comment:
why do you need to do this, arma automatically removes munitions from pylons when firing already?

Reply:
because arma decided that it didn't want to remove the pylon when it fired

Reply:
that's because the pylons should not be removed. a pylon is a launch rail.
the missile model should only be set on the ammo, the pylon model should be a launch rail or nothing